### DISCUSSION TITLE
How to define a function/method in Rust which accepts at most 2 positional arguments?

### DISCUSSION THREAD

**Discussion by lycantropos** [PyO3/pyo3#1913] [Questions] [answered]
What is an idiomatic way of defining a Rust function which accepts limited number of positional arguments?
E.g., I want to be able to write something like
```python
>>> from rust_module import foo
>>> foo()
>>> foo(1)
>>> foo(1, 2)
>>> foo(1, 2, 3)
Traceback (most recent call last):
  File "<input>", line 1, in <module>
TypeError: foo() takes from 0 to 2 positional arguments but 3 were given
```
In Python C API we can use [`PyArg_ParseTuple`](https://docs.python.org/3/c-api/arg.html#c.PyArg_ParseTuple), but are there any examples of how to do it properly in PyO3?

**Comment by birkenfeld** [comment, marked as answer]
Like in Python - just take two arguments with one having a default value.

Or take `*args` and extract the contents yourself, this is closer to the C and using `PyArg_UnpackTuple` (which you probably meant).
